src/syscheckd/fim_db.c

```diff
@@ -1138,33 +1145,39 @@ void fim_db_remove_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex
         w_mutex_unlock(mutex);
 }
 
-void fim_db_process_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex, void *arg) {
+void fim_db_process_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex, void *fim_ev_mode, void *w_evt) {
 
-    fim_event_mode mode = (fim_event_mode) arg;
+    fim_event_mode mode = (fim_event_mode) fim_ev_mode;
     int conf_file = fim_configuration_directory(entry->path, "file");
 
     switch (mode) {
+        /*
+            Don't send alert if received mode and mode in configuration aren't the same
+        */
 
         case FIM_REALTIME:
             if (!(syscheck.opts[conf_file] & REALTIME_ACTIVE)){
+                fim_db_remove_path(fim_sql, entry, mutex, (void *) (int) false, (void *) mode, w_evt);
                 return;
             }
             break;
 
         case FIM_WHODATA:
             if (!(syscheck.opts[conf_file] & WHODATA_ACTIVE)) {
+                fim_db_remove_path(fim_sql, entry, mutex, (void *) (int) false, (void *) mode, w_evt);
                 return;
             }
             break;
 
         case FIM_SCHEDULED:
             if (!(syscheck.opts[conf_file] & SCHEDULED_ACTIVE)) {
+                fim_db_remove_path(fim_sql, entry, mutex, (void *) (int) false, (void *) mode, w_evt);
                 return;
             }
             break;
     }
 
-    fim_db_remove_path(fim_sql, entry, mutex, (void *) (int) 1);
+    fim_db_remove_path(fim_sql, entry, mutex, (void *) (int) true, (void *) mode, w_evt);
```

Comment:
Please define two constants to name that parameter as alert or non-alert.

Reply:
Fixed 6cd233d